src/components/admin/UserProfileModal/UserProfileModal.tsx

```diff
@@ -0,0 +1,67 @@
+import { Fragment, useRef } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+
+import { STRING_ZERO_WIDTH_SPACE } from "settings";
+
+import { useProfileModalControls } from "hooks/useProfileModalControls";
+
+import { UserProfileModalBody } from "./components/UserProfileModalBody";
+
+export const UserProfileModal = () => {
+  const {
+    selectedUserId,
+    hasSelectedProfile,
+    closeUserProfileModal,
+  } = useProfileModalControls();
+
+  const cancelButtonRef = useRef(null);
+
+  return (
+    <Transition.Root show={hasSelectedProfile} as={Fragment}>
+      <Dialog
+        as="div"
+        className="AdminProfileModal fixed z-10 inset-0 overflow-y-auto"
+        initialFocus={cancelButtonRef}
+        onClose={closeUserProfileModal}
+      >
+        <div className="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0">
+          <Transition.Child
+            as={Fragment}
+            enter="ease-out duration-300"
```

Comment:
Maybe we should extract these as variables and import them.

Reply:
extracted: https://github.com/sparkletown/sparkle/pull/2931/commits/493bfda6c66a90eb3c85efbc5943a9e57dacef93